### PR TITLE
mds: Reorganize structure members in snap header

### PR DESCRIPTION
**Pull request by varshar16**
Fixes: https://tracker.ceph.com/issues/43425
Signed-off-by: Varsha Rao <varao@redhat.com>

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard backend`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

File: src/mds/snap.h [hunk 1]
```diff
@@ -20,91 +20,83 @@
...
-  snapid_t current_parent_since;
-  map<snapid_t, SnapInfo> snaps;
-  map<snapid_t, snaplink_t> past_parents;  // key is "last" (or NOSNAP)
-  set<snapid_t> past_parent_snaps;
-
-  __u32 flags;
-  enum {
-    PARENT_GLOBAL = 1 << 0,
-  };
+  sr_t() {}
```
Comment (batrick) [inline src/mds/snap.h]: We can drop the constructor now (and use the default)

File: src/mds/snap.h [hunk 2]
```diff
@@ -20,91 +20,83 @@
...
 WRITE_CLASS_ENCODER(sr_t)
 
+class CInode;
+class MDCache;
+
+inline bool operator==(const SnapInfo &l, const SnapInfo &r)
+{
+  return l.snapid == r.snapid && l.ino == r.ino &&
+	 l.stamp == r.stamp && l.name == r.name;
+}
```
Comment (batrick) [inline src/mds/snap.h]: These inline functions should stay beneath the class they're related to.

File: src/mds/snap.h [hunk 3]
```diff
@@ -20,91 +20,83 @@
...
+
+  __u32 flags = 0;
+  enum {
+    PARENT_GLOBAL = 1 << 0,
+  };
 };
 WRITE_CLASS_ENCODER(sr_t)
 
+class CInode;
+class MDCache;
```
Comment (batrick) [inline src/mds/snap.h]: We can drop these two lines I think?

Comment (varshar16) [reply src/mds/snap.h]: MDCache is required. I have removed CInode.

File: src/mds/snap.h (hunk 2 above)
Comment (varshar16) [reply src/mds/snap.h]: Done

File: src/mds/snap.h (hunk 1 above)
Comment (varshar16) [reply src/mds/snap.h]: Done

**Pull request merged**